### PR TITLE
[Enterprise Search] Add delete connector sync job by id endpoint

### PR DESCRIPTION
**Pull request by timgrein**
This PR adds an endpoint to delete a connector sync job by id. 
I've moved the `_connector/_sync_job` constant now to `EnterpriseSearch` as it's used at multiple places.

### CODE REVIEW

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/DeleteConnectorSyncJobAction.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    public static final String NAME = "cluster:admin/xpack/connector/sync_job/delete";
+
+    private DeleteConnectorSyncJobAction() {
+        super(NAME, AcknowledgedResponse::readFrom);
+    }
+
+    public static class Request extends ActionRequest implements ToXContentObject {
+        public static final String EMPTY_CONNECTOR_SYNC_JOB_ID_ERROR_MESSAGE =
+            "[connector_sync_job_id] of the connector sync job cannot be null or empty.";
+        public static final ParseField CONNECTOR_SYNC_JOB_ID_FIELD = new ParseField("connector_sync_job_id");
```
Comment (jedrazb) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/DeleteConnectorSyncJobAction.java:41]: nit: this is likely to be reused by other requests, should we put it to `SyncJob` class?  Can be addressed in the next PR

File: x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexServiceTests.java [hunk 2]
```diff
@@ -151,6 +154,24 @@ public void testCreateConnectorSyncJob_WithMissingConnectorId_ExpectException()
...
+            connector.getConnectorId()
+        );
+        PostConnectorSyncJobAction.Response response = awaitPutConnectorSyncJob(syncJobRequest);
+        String syncJobId = response.getId();
+
+        assertThat(syncJobId, notNullValue());
+
+        DeleteResponse deleteResponse = awaitDeleteConnectorSyncJob(syncJobId);
+
+        assertThat(deleteResponse.status(), equalTo(RestStatus.OK));
```
Comment (jedrazb) [inline x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexServiceTests.java:168]: nit/todo: once the `get` endpoint is implemented, we should try querying for the doc with that id

**Review by carlosdelest — APPROVED** [review]
Way to go! 💯

**Review by jedrazb — APPROVED** [review]
LGTM :rocket:

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/DeleteConnectorSyncJobAction.java (hunk 1 above)
Comment (timgrein) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/DeleteConnectorSyncJobAction.java:41]: Yeah I think we could've a `Connectors` class maybe similar to `EnterpriseSearch`, which holds some of these constants. Definitely open to discuss.

**Comment by timgrein** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-unix-sample

**Pull request merged**